### DISCUSSION TITLE
Usage of swiftshader_indirect for GMD on GitHub Actions

### DISCUSSION THREAD

**Discussion by eygraber** [android/nowinandroid#1693] [Q&A]
I see that the [NightlyBaselineProfiles](https://github.com/android/nowinandroid/blob/main/.github/workflows/NightlyBaselineProfiles.yaml#L58C65-L58C85) sets

> android.testoptions.manageddevices.emulator.gpu="swiftshader_indirect"

even though [hardware acceleration is used](https://github.com/android/nowinandroid/blob/main/.github/workflows/NightlyBaselineProfiles.yaml#L21).

Are there still performance benefits to using `swiftshader_indirect`?

**Comment by yschimke** [comment]
It's also still the guidance on android-emulator-runner https://github.com/ReactiveCircus/android-emulator-runner/blob/main/README.md

**Reply by eygraber** [reply]
It wasn't clear to me if that was something they needed to update, or if it's still valid - https://github.com/ReactiveCircus/android-emulator-runner/discussions/419

**Reply by yschimke** [reply]
Relevant discussion on https://issuetracker.google.com/issues/287312019

**Reply by eygraber** [reply]
What's confusing me is that the reply to [this comment (yours?)](https://issuetracker.google.com/issues/287312019#comment40) from that issue says to use `swiftshader_indirect`, but [the docs](https://developer.android.com/studio/run/emulator-acceleration#command-gpu) seem to indicate that `host` will perform the best if hardware acceleration is available. 

> swiftshader_indirect: Use a Quick Boot-compatible variant of [SwiftShader](https://swiftshader.googlesource.com/SwiftShader) to render graphics using software acceleration. This option is a good alternative to host mode if your computer can't use hardware acceleration.

**Reply by yschimke** [reply]
Maybe worth a reply on that issue?

**Comment by alamryaqdh-sketch** [comment]
> I see that the [NightlyBaselineProfiles](https://github.com/android/nowinandroid/blob/main/.github/workflows/NightlyBaselineProfiles.yaml#L58C65-L58C85) sets
> 
> > android.testoptions.manageddevices.emulator.gpu="swiftshader_indirect"
> 
> even though [hardware acceleration is used](https://github.com/android/nowinandroid/blob/main/.github/workflows/NightlyBaselineProfiles.yaml#L21).
> 
> Are there still performance benefits to using `swiftshader_indirect`?
